package/gluon-mesh-vpn-wireguard/files/usr/bin/wg-broker-server

```diff
@@ -0,0 +1,210 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-2.0
+#
+# Copyright (C) Christof Schulze <christof@christofschulze.com>. All Rights Reserved.
+#
+
+BROKERPORT=40000	# tcp port where the broker listens for connections
+STARTPORT=40000		# start of udp portrange accepting wireguard connections
+ENDPORT=41000		# end of udp portrange accepting wireguard connections
+BABELPORT=33123		# control port of local babeld
+MTU=1374		# MTU of VPN in this network
+WG="wg"			# wireguard command
+WAN="br-wan"		# accepting inbound wireguard connections on this interface
+MAXCONNECTIONS=150	# allow MAXCONNECTIONS concurrent vpn connections
+
+[[ -f /etc/wg-broker/config ]] && . /etc/wg-broker/config
+
+[[ -f /etc/wg-broker/secret ]] || {
+  echo expecting to find wg secret key in /etc/wg-broker/secret >&2
+  exit 1
+}
+
+PRIVATEKEY=/etc/wg-broker/secret
+
+set -x
+get_wg_interfaces() {
+    $WG show interfaces |grep babel-wg
+}
+
+get_connection_count() {
+    interfaces=( $(get_wg_interfaces) )
+    echo ${#interface[@]}
+}
+
+get_port() {
+    $WG show ${1:-all} listen-port
+}
+
+port_is_assigned() {
+    get_port | grep -q "$1"
+}
+
+find_free_port() {
+    local port=$((STARTPORT + $RANDOM%1000)) # adding RANDOM to  not assign ports sequentially and save some time in below loop
+    local sport=$port #
+
+    while $(port_is_assigned $port)
+    do
+	port=$((port+1))
+	[[ $port -eq $ENDPORT ]] && port=$STARTPORT
+	[[ $port -eq $sport ]] &&
+	{
+	  echo 0
+	  return
+	}
+    done
+
+    echo $port
+}
+
+add_interface() {
+    local newdevice="$1"
+    local freeport="$2"
+    local public_key="$3"
+    local rport="$4"
+    local remoteip="$5"
+
+    ip link add dev $newdevice type wireguard
+    ip link set multicast on mtu $MTU dev $newdevice
+    ip a a fe80::1/64 dev $newdevice
```

Comment:
This will prevent direct links between two wg-broker-server instances.

Reply:
I disagree. the server-part of the connection always has that ip address. It does not mean there can be a client-part having a different IP-address.

Reply:
Why should it always have to have that address? If you are running a routing protocol like babel over it, then it doesn't matter anyway.

Reply:
It needs to have *some* address and wireguard stopped generating random fe80s a while back. There isn't much need for a random one on a PTP link either way.